src/gwt/src/org/rstudio/studio/client/workbench/views/environment/EnvironmentPane.java

```diff
@@ -150,9 +170,38 @@ protected Toolbar createMainToolbar()
 
    @Override
    protected SecondaryToolbar createSecondaryToolbar()
+   {
+      initSecondaryToolbar();
+      return secondaryToolbar_;
+   }
+   
+   private void initSecondaryToolbar()
    {
       SecondaryToolbar toolbar = new SecondaryToolbar("Environment Tab Second");
       
+      languageMenu_ = new ToolbarPopupMenu();
+      
+      MenuItem rMenuItem = new MenuItem("R", () -> setActiveLanguage("R", true));
+      languageMenu_.addItem(rMenuItem);
+      
+      MenuItem pyMenuItem = new MenuItem("Python", () -> setActiveLanguage("Python", true));
+      languageMenu_.addItem(pyMenuItem);
+      
+      languageButton_ = new ToolbarMenuButton(
+            "R",
+            ToolbarButton.NoTitle,
+            (ImageResource) null,
+            languageMenu_);
+      
+      // don't show the language button at first; instead,
+      // toggle its visibility after the widget is created
+      // depending on whether Python has been initialized
```

Comment:
Not a checkin blocker, but I wonder if it'd be nicer to show this all the time as a sort of advertisement of what's possible. Attempting to switch modes before Python is initialized could show instructions for enabling the feature (by initializing python or installing the appropriate software). What do you think?

Reply:
At first I was opposed, but now I think this is worth doing -- especially so now because the trigger that makes this button visible is non-obvious (`reticulate` must be loaded _and_ Python must be initialized). I'll make that change.